segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/DebugStreamSegmentContainer.java

```diff
@@ -58,4 +66,27 @@
         ArrayView segmentInfo = MetadataStore.SegmentInfo.recoveredSegment(streamSegmentName, length, isSealed);
         return metadataStore.createSegment(streamSegmentName, segmentInfo, new TimeoutTimer(TIMEOUT));
     }
+
+    @Override
+    public CompletableFuture<Void> copySegment(Storage storage, String sourceSegment, String targetSegment, ExecutorService executor) {
+        val targetHandle = storage.create(targetSegment, TIMEOUT).thenCompose(v -> storage.openWrite(targetSegment)).join();
```

Comment:
Need to get rid of these joins as well.
It should look something like below .
Also you don't need to openWrite again after create.
```
return storage.create(targetSegment, TIMEOUT).thenCompose(targetHandle -> {
           return storage.getStreamSegmentInfo(sourceSegment, TIMEOUT).thenCompose(info -> {
               return storage.openRead(sourceSegment).thenCompose(sourceHandle  ->  {
                      offset = 0;
                      return Futures.loop( .....
              }
        })
}})
```

Reply:
Done.